### PR TITLE
MAT-180 XSS Handling for Measure Names in Grids

### PR DESCRIPTION
**Pull request by RohitKandimalla**

### CODE REVIEW

File: src/main/java/mat/client/shared/MeasureLibraryResultTable.java [hunk 1]
```diff
@@ -238,12 +238,12 @@ private SafeHtml getMeasureNameColumnToolTip(ManageMeasureSearchModel.Result obj
...
                     .appendHtmlConstant("<button id='div1' type=\"button\" title=\""
-                            + object.getName() + "\" tabindex=\"-1\" class=\" " + cssClass + "\"></button>")
+                            + SafeHtmlUtils.htmlEscape(object.getName()) + "\" tabindex=\"-1\" class=\" " + cssClass + "\"></button>")
                     .appendHtmlConstant("</div>");
 
         }
         sb.appendHtmlConstant("<div class=\"pull-left\" title=\" Click to open "
-                + object.getName() + "\" tabindex=\"0\">" + object.getName() + "</div>");
+                + SafeHtmlUtils.htmlEscape(object.getName()) + "\" tabindex=\"0\">" + SafeHtmlUtils.htmlEscape(object.getName()) + "</div>");
         return sb.toSafeHtml();
```
Comment (serhii-ilin) [inline src/main/java/mat/client/shared/MeasureLibraryResultTable.java]: btw. it already has sb.toSafeHtml. Does it do the same thing as htmlEscape?

Comment (RohitKandimalla) [reply src/main/java/mat/client/shared/MeasureLibraryResultTable.java]: appendHtmlConstant will take in HTML content without escaping special chars.

Comment (serhii-ilin) [reply src/main/java/mat/client/shared/MeasureLibraryResultTable.java]: but now about "sb.toSafeHtml()"

Comment (RohitKandimalla) [reply src/main/java/mat/client/shared/MeasureLibraryResultTable.java]: toSafeHtml doesn't escape special chars, it Returns the safe HTML accumulated in the builder 'sb'. sb.appendEscaped Appends a string after HTML-escaping it.

Comment (serhii-ilin) [reply src/main/java/mat/client/shared/MeasureLibraryResultTable.java]: good to know. thank you

**Pull request merged**